src/main/kotlin/com/linchpino/core/service/InterviewService.kt

```diff
@@ -72,4 +79,14 @@ class InterviewService(
             entity.jobSeekerAccount?.email,
         )
     }
+
+    @Transactional(readOnly = true)
+    fun upcomingInterviews(authentication: Authentication, page: Pageable): Page<InterviewListResponse> {
+        val email = when (authentication) {
+            is JwtAuthenticationToken -> authentication.name
+            is BearerTokenAuthentication -> (authentication.principal as OAuth2IntrospectionAuthenticatedPrincipal).name
+            else -> throw UnsupportedOperationException("Authentication type not supported")
```

Comment:
Are we have any test case to getting UnsupportedOperationException("Authentication type not supported") ?